tests/test_processors.py

```diff
@@ -159,3 +184,12 @@ def test_do_decode_update():
 
     processor.block = bk_block
     assert processor.check_update(None, old_asset, "") is False
+
+
+def test_metadata_state_update():
```

Comment:
not sure you need this test. This processor is much simpler than the other ones, and the functional tests (one for deleting and one for re-creating) will cover all cases and provide enough coverage. This file is more for testing the "off-cases" of the processors.